daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs

```diff
@@ -551,6 +554,38 @@ codegenTests codegenDir damlTypes = testGroup "daml codegen" (
                         contents <- listDirectory outDir
                         assertBool "bindings were written" (not $ null contents)
 
+createDamlAppTests :: TestTree
+createDamlAppTests = testGroup "create-daml-app" [gettingStartedGuideTest | not isWindows]
+  where
+    gettingStartedGuideTest = testCase "Getting Started Guide" $
+      withTempDir $ \tmpDir -> do
+        let tsLibs = ["daml-types", "daml-ledger", "daml-react"]
+        forM_  tsLibs $ \tsLib -> do
+          srcDir <- locateRunfiles $ mainWorkspace </> "language-support" </> "ts" </> tsLib </> "npm_package"
+          copyDirectory srcDir (tmpDir </> tsLib)
+        BSL.writeFile (tmpDir </> "package.json") $ Aeson.encode $ object
+          [ "private" .= True
+          , "workspaces" .= ["create-daml-app/daml-ts", "create-daml-app/ui" :: String]
```

Comment:
This is slightly different to the manual test procedure - do we not want the package.json inside the create-daml-app dir created after `daml new`?

Reply:
IMO, this is an improvement of the manual test procedure we discussed yesterday. We can even have the `package.json` we use to trick `yarn` into using our packages outside of the project. Doesn't that sound better?

Reply:
Not sure if it's better or worse, but I didn't know that this package.json would have an effect here since you call `daml codegen ts` in the `create-daml-app` directory.

Reply:
`daml codegen ts` itself does not care about this itself at all. It calls `yarn install` in `daml-ts`, which then does its magic and searches for a workspace `package.json` somewhere along the path to the root directory.

The advantage of what we're doing here is that we don't need to mess around with the directory produced by `daml new` at all. The only thing we mess around with is tricking `yarn install` into picking up our libraries. In fact, this approach can be used to share a single `node_modules` folder as a cache between multiple test runs without weirdly copying it around. I already have big plans for using this to make the `daml2ts` tests faster.

Reply:
> I already have big plans for using this to make the `daml2ts` tests faster.

I’ll only approve if they get faster by at least 30x.

Reply:
We should only measure the sum of all `yarn install` steps for this benchmark. If this change doesn't make it 30x fast, then I'll add more test cases until it does make it 30x faster. And I'll add them each in a separate PR. That's what I call two birds with one stone. 🤓